src/main/java/bio/terra/app/usermetrics/UserMetricsInterceptor.java

```diff
@@ -47,14 +48,18 @@ public void afterCompletion(
       throws Exception {
     String method = request.getMethod().toUpperCase();
     String path = request.getRequestURI();
-    AuthenticatedUserRequest userRequest = authenticatedUserRequestFactory.from(request);
-    System.err.println("Start client " + bardClient);
+    AuthenticatedUserRequest userRequest;
+    try {
+      userRequest = authenticatedUserRequestFactory.from(request);
+    } catch (UnauthorizedException e) {
+      // Don't track unauthenticated requests
+      return;
+    }
     // Don't log metrics if bard isn't configured, the path is part of the ignore list or the
     // request is unauthorized
     if (StringUtils.isEmpty(metricsConfig.getBardBasePath())
         || ignoreEventForPath(path)
         || StringUtils.isEmpty(userRequest.getToken())) {
```

Comment:
I'm wondering how it's possible to get here without a token.

Reply:
Good point...now that the auth check happens above effectively, we don't need that.  I'll remove